code/game/objects/items/weapons/twohanded.dm

```diff
@@ -303,8 +303,97 @@
 	hitsound = 'sound/weapons/bladeslice.ogg'
 	attack_verb = list("attacked", "poked", "jabbed", "torn", "gored")
 	sharpness = IS_SHARP
+	var/explosive = 0
+	var/war_cry = "AAAAARGH!"
 
 /obj/item/weapon/twohanded/spear/update_icon()
-	icon_state = "spearglass[wielded]"
-	return
+	if(explosive)
+		icon_state = "spearbomb[wielded]"
+		return
+	else
+		icon_state = "spearglass[wielded]"
+		return
+
+/obj/item/weapon/twohanded/spear/afterattack(atom/movable/AM, mob/user, proximity)
+	if(!proximity)
+		return
+	switch(explosive)

```

Comment:
You're crushing my dreams of spawning a ton of C4 as gang boss and sending my war boys to valhalla